packages/vscode-extension/src/devices/preview.ts

```diff
@@ -37,7 +37,12 @@ export class Preview implements Disposable {
 
       const streamURLRegex = /(http:\/\/[^ ]*stream\.mjpeg)/;
 
-      lineReader(subprocess).onLineRead((line) => {
+      lineReader(subprocess).onLineRead((line, stderr) => {
+        if (stderr) {
+          Logger.warn("sim-server err:", line);
+          return;
+        }
```

Comment:
I'm a bit skeptical about this, simserver runs a few processes concurrently and one of them can log to `stderr` a thing that is unrelated to the `stdout` output that is awaited by the IDE in the meantime. Loggers by default output to `stderr`, even if a message is for informational/debugging purposes, so I wouldn't assume that something appearing on `stderr` means an error occured.

What I'd suggest is expecting the communication from simserver on `stdout` (responses to commands passed on `stdin` like `preview_ready`, `replay_ready`, `replay_error`) and ignore `stderr`, leaving it for simserver's logging purposes. It would require no changes in this repo, on simserver's side I'd bring back the default logging to stdout and replace `log::info!` calls with `println!` when a message is a command response.

Reply:
I understand, however it is still important that we read stderr and forward that to the logs. For example the recent crash that we were seeing due to unresolved shared library was something that got printed out by the system into stderr. We didn't see that in the extension logs at all because of that.